### PR TITLE
Card fix

### LINKED ISSUE #588: card: All CSS classes for content layouts have been removed

### PR DESCRIPTION
**Pull request by prateekbh**
Fixes #588 #586 related to card.

### CODE REVIEW

File: Card/index.d.ts [hunk 1]
```diff
@@ -1,43 +1,33 @@
+import Button from '../Button';
+import Icon from '../Icon';
 import MaterialComponent from '../MaterialComponent';
 import { VNode } from 'preact';
-import Button from '../Button';
 
 declare class Section<PropsType = {}, StateType = {}> extends MaterialComponent<PropsType & JSX.HTMLAttributes, StateType> {}
```
Comment (YePpHa) [inline Card/index.d.ts]: Not used anymore

File: Card/index.d.ts [hunk 2]
```diff
@@ -1,43 +1,33 @@
...
+  static ActionButton: typeof ActionButton;
+  static ActionIcon: typeof ActionIcon;
   static Media: typeof Media;
-  static Title: typeof Title;
-  static Subtitle: typeof Subtitle;
-  static HorizontalBlock: typeof HorizontalBlock;
-  static MediaItem: typeof MediaItem;
 }
 
 declare class Primary extends Section<JSX.HTMLAttributes, {}> {}
```
Comment (YePpHa) [inline Card/index.d.ts]: Not used anymore

File: Card/index.d.ts [hunk 3]
```diff
@@ -1,43 +1,33 @@
...
 declare class SupportingText extends Section<JSX.HTMLAttributes, {}> {}
```
Comment (YePpHa) [inline Card/index.d.ts]: Not used anymore

File: Card/index.d.ts [hunk 4]
```diff
@@ -1,43 +1,33 @@
...
-  static SupportingText: typeof SupportingText;
   static Actions: typeof Actions;
-  static Action: typeof Action;
+  static ActionButton: typeof ActionButton;
+  static ActionIcon: typeof ActionIcon;
   static Media: typeof Media;
-  static Title: typeof Title;
-  static Subtitle: typeof Subtitle;
-  static HorizontalBlock: typeof HorizontalBlock;
-  static MediaItem: typeof MediaItem;
```
Comment (YePpHa) [inline Card/index.d.ts]: Missing `CardMediaContent`

File: Card/index.d.ts [hunk 5]
```diff
@@ -1,43 +1,33 @@
...
-declare class Action extends Button<JSX.HTMLAttributes, {}> {}
-
-declare interface ITitleProps extends JSX.HTMLAttributes {
-  large?: boolean;
-}
-declare class Title extends MaterialComponent<ITitleProps, {}> {}
+declare class ActionButton extends Button<JSX.HTMLAttributes, {}> {}
+declare class ActionIcon extends Icon<JSX.HTMLAttributes, {}> {}
 
 declare class Subtitle extends MaterialComponent<JSX.HTMLAttributes, {}> {}
```
Comment (YePpHa) [inline Card/index.d.ts]: Not used anymore

File: Card/index.d.ts [hunk 6]
```diff
@@ -1,43 +1,33 @@
...
 declare class HorizontalBlock extends Section<JSX.HTMLAttributes, {}> {}
```
Comment (YePpHa) [inline Card/index.d.ts]: Not used anymore

**Comment by prateekbh** [comment]
@YePpHa fixed your comments... please verify

File: Card/index.d.ts (hunk 4 above)
Comment (prateekbh) [reply Card/index.d.ts]: Added

Comment (YePpHa) [reply Card/index.d.ts]: You forgot to add it to Card as `static CardMediaContent: typeof CardMediaContent`.

File: Card/index.d.ts [hunk 7]
```diff
@@ -1,43 +1,28 @@
...
-declare class Subtitle extends MaterialComponent<JSX.HTMLAttributes, {}> {}
-declare class HorizontalBlock extends Section<JSX.HTMLAttributes, {}> {}
+declare class Actions extends MaterialComponent<IActionsProps, {}> {}
+declare class ActionButton extends Button<JSX.HTMLAttributes, {}> {}
+declare class ActionIcon extends Icon<JSX.HTMLAttributes, {}> {}
+declare class CardMediaContent extends MaterialComponent<IActionsProps, {}> {}
 
-declare interface IMediaItemProps extends JSX.HTMLAttributes {
-  x?: "1dot5"|"2"|"3";
+declare interface IMediProps extends JSX.HTMLAttributes {
```
Comment (YePpHa) [inline Card/index.d.ts]: Please fix spelling error `IMediProps` should be `IMediaProps`.

File: Card/index.d.ts [hunk 8]
```diff
@@ -1,43 +1,28 @@
...
+  x?: "square"|"16-9";
 }
-declare class MediaItem extends MaterialComponent<IMediaItemProps, {}> {}
\ No newline at end of file
+declare class Media extends MaterialComponent<IMediProps, {}> {}
```
Comment (YePpHa) [inline Card/index.d.ts]: Please fix spelling error `IMediProps` should be `IMediaProps`.

File: Card/index.d.ts [hunk 9]
```diff
@@ -1,43 +1,28 @@
...
-  static SupportingText: typeof SupportingText;
   static Actions: typeof Actions;
-  static Action: typeof Action;
+  static ActionButton: typeof ActionButton;
+  static ActionIcon: typeof ActionIcon;
   static Media: typeof Media;
-  static Title: typeof Title;
-  static Subtitle: typeof Subtitle;
-  static HorizontalBlock: typeof HorizontalBlock;
-  static MediaItem: typeof MediaItem;
```
Comment (YePpHa) [inline Card/index.d.ts]: You forgot to add it to Card as `static CardMediaContent: typeof CardMediaContent`.

**Comment by prateekbh** [comment]
@YePpHa please review

**Comment by prateekbh** [comment]
@YePpHa Docs and sample will be a separate PR altogether

**Pull request merged**